### ISSUE TITLE
core-configuration-layer.el depend on deprecated `defmethod` from EIEIO

### ISSUE THREAD

**Issue by oxyum** [syl20bnr/spacemacs#15230]
EIEIO was deprecated for years now. Most spacemacs codebase was migrated to `cl-defmethod`, but not `core-configuration-layer.el`.

But at the end of 2021 `eieio-compat.el` was finally killed from emacs repo:
https://github.com/emacs-mirror/emacs/commit/d65534d254b3965ea82a9300c12c5c07f88818b7

So spacemacs became unusable on newer emacs.

**Comment by sunlin7** [comment]
And the dependency package `multi-line` in https://github.com/syl20bnr/spacemacs/blob/0206197b224f67e6a4c65bb850758f4affc9cc84/layers/%2Bspacemacs/spacemacs-editing/packages.el#L46  also require the `defmethod`, and sadly the package `multi-line` is in-actived https://github.com/IvanMalison/multi-line.

**Comment by lebensterben** [comment]
the problem with `multi-line` is fixed in an upstream PR https://github.com/IvanMalison/multi-line/pull/16
I will take a look on core-configuration-layer.el

**Comment by cormacc** [comment]
While waiting for that PR to get merged, I've made the following edit locally to spacemacs-editing/packages.el:

```
        ;; multi-line
        ;; Workaround pending merge of PR https://github.com/IvanMalison/multi-line/pull/16
        (multi-line :location (recipe
                            :fetcher github
                            :repo "eklitzke/multi-line"
                            :branch "master"))
```

**Comment by oxyum** [comment]
fixed in #15238
